cloud/amazon/ec2_elb_facts.py

```diff
@@ -0,0 +1,198 @@
+            'instance_port': listener[1],
+            'protocol': listener[2],
+        }
+        try:
+            ssl_certificate_id = listener[4]
+        except IndexError:
+            pass
+        else:
+            if ssl_certificate_id:
+                listener_dict['ssl_certificate_id'] = ssl_certificate_id
+        listener_list.append(listener_dict)
+
+    return listener_list
+
+
+def get_health_check(health_check):
+    protocol, port_path = health_check.target.split(':')
+    try:
+        port, path = port_path.split('/')
+        path = '/{}'.format(path)
+    except ValueError:
+        port = port_path
+        path = None
+
+    health_check_dict = {
+        'ping_protocol': protocol.lower(),
+        'ping_port': int(port),
+        'response_timeout': health_check.timeout,
+        'interval': health_check.interval,
+        'unhealthy_threshold': health_check.unhealthy_threshold,
+        'healthy_threshold': health_check.healthy_threshold,
+    }
+    if path:
+        health_check_dict['ping_path'] = path
+    return health_check_dict
+
+
+def get_elb_info(elb):
+    elb_info = {
+        'name': elb.name,
+        'zones': elb.availability_zones,
+        'dns_name': elb.dns_name,
+        'instances': [instance.id for instance in elb.instances],
+        'listeners': get_elb_listeners(elb.listeners),
+        'scheme': elb.scheme,
+        'security_groups': elb.security_groups,
+        'health_check': get_health_check(elb.health_check),
+        'subnets': elb.subnets,
+    }
+    if elb.vpc_id:
+        elb_info['vpc_id'] = elb.vpc_id
+
+    return elb_info
+
+
+def list_elb(connection, module):
+    elb_names = module.params.get("names")
+    if not elb_names:
+        elb_names = None
+
+    try:
+        all_elbs = connection.get_all_load_balancers(elb_names)
+    except BotoServerError as e:
+        module.fail_json(msg=get_error_message(e.args[2]))
+
+    elb_array = []
+    for elb in all_elbs:
+        elb_array.append(get_elb_info(elb))
+
+    module.exit_json(elbs=elb_array)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            names={'default': None, 'type': 'list'}
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+
+    if region:
+        try:
+            connection = connect_to_aws(boto.ec2.elb, region, **aws_connect_params)
+        except (boto.exception.NoAuthHandlerFound, StandardError), e:
+            module.fail_json(msg=str(e))
+    else:
+        module.fail_json(msg="region must be specified")
+
+    list_elb(connection, module)
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.ec2 import *
+
+# this is magic, see lib/ansible/module_common.py
+#<<INCLUDE_ANSIBLE_MODULE_COMMON>>
```

Comment:
Ah, I thought that was the case now-a-days I just picked a module that hasn't been updated since the change I suppose. Fixed in https://github.com/mjschultz/ansible-modules-extras/commit/913266c04f62b775f30d82ffb8c163d0b77c24fa.